app/controllers/providers/edit_initial_allocations_controller.rb

```diff
@@ -0,0 +1,125 @@
+module Providers
+  class EditInitialAllocationsController < ApplicationController
+    def do_you_want
+      if request.post?
+        case params[:allocation][:request_type]
+        when AllocationsView::RequestType::INITIAL
+          redirect_to number_of_places_provider_recruitment_cycle_allocation_path(
+            provider_code: allocation.accredited_body.provider_code,
+            recruitment_cycle_year: recruitment_cycle.year,
+            training_provider_code: allocation.provider.provider_code,
+            id: allocation.id,
+          )
+        when AllocationsView::RequestType::DECLINED
+          destroy
+          redirect_to confirm_deletion_provider_recruitment_cycle_allocation_path(
+            provider_code: provider.provider_code,
+            recruitment_cycle_year: recruitment_cycle.year,
+            training_provider_code: training_provider.provider_code,
```

Comment:
On line 10 above, we get the training provider code like this:

```ruby
            training_provider_code: allocation.provider.provider_code,
```

Which makes sense because it avoids doing a request to the API. Would it make sense to do that here too?

Reply:
Yes, makes sense to make this standardise across all these changes actually